src/StatsdClient/Statsd.cs

```diff
@@ -51,14 +51,28 @@ public abstract class Metric : ICommandType
                 string unit = _commandToUnit[typeof(TCommandType)];
                 var allTags = ConcatTags(constantTags, tags);
 
-                return string.Format(
-                    CultureInfo.InvariantCulture,
-                    "{0}:{1}|{2}{3}{4}",
-                    full_name,
-                    value,
-                    unit,
-                    sampleRate == 1.0 ? "" : string.Format(CultureInfo.InvariantCulture, "|@{0}", sampleRate),
-                    allTags);
+                if (value is double)
+                {
+                    return string.Format(
+                            CultureInfo.InvariantCulture,
+                            "{0}:{1:G15}|{2}{3}{4}",
+                            full_name,
+                            value,
+                            unit,
+                            sampleRate == 1.0 ? "" : string.Format(CultureInfo.InvariantCulture, "|@{0}", sampleRate),
+                            allTags);
+                }
+                else
+                {
+                    return string.Format(
+                            CultureInfo.InvariantCulture,
+                            "{0}:{1}|{2}{3}{4}",
+                            full_name,
+                            value,
+                            unit,
+                            sampleRate == 1.0 ? "" : string.Format(CultureInfo.InvariantCulture, "|@{0}", sampleRate),
+                            allTags);
+                }

```

Comment:
The new behavior in .Net Core is a fix and using `{1:G15}` is a breaking change requiring a major bump version.

I think the unit tests should be fixed:
```c#
        public void gauge_double_rounding()
        {
            var value = (double)1 / 9;
            _dogStatsdService.Gauge("gauge", value);
            AssertWasReceived($"gauge:{value}|g");
        }
```